synapse/util/caches/descriptors.py

```diff
@@ -37,6 +39,12 @@
 logger = logging.getLogger(__name__)
 
 
+cache_pending_metric = Gauge(
+    "synapse_util_caches_cache_pending",
```

Comment:
so, question. The existing cache metrics (https://github.com/matrix-org/synapse/blob/b3145741072e5cfe49d4fd70161864e623824970/synapse/util/caches/__init__.py#L42 etc) have a colon in their names, but the prometheus exposition code duplicates them, replacing colons with underscores, going so far as to call colons weird: https://github.com/matrix-org/synapse/blob/b3145741072e5cfe49d4fd70161864e623824970/synapse/metrics/_exposition.py#L152

I'm assuming that means that the colon-names are deprecated and underscores are the new black, but I'm not sure of any authority or conscious decision here.